zerver/tests/test_upload.py

```diff
@@ -203,6 +203,21 @@ def test_file_upload_authed(self) -> None:
         self.send_stream_message(self.example_user("hamlet"), "Denmark", body, "test")
         self.assertIn('title="zulip.txt"', self.get_last_message().rendered_content)
 
+        # Now try the format that's supposed to return 200s
+        result = self.client_get(uri + "?url_only=true")
+        self.assert_json_success(result)
+        data = result.json()
+        url_only_url = data['url']
+        self.assertIn('user_uploads/local/', url_only_url)
+        # The generated url has a token authorizing the requestor to access the file
+        # without being logged in.
+        self.logout()
+        self.assert_url_serves_contents_of_file(url_only_url, b"zulip!")
```

Comment:
Let's add a final check that the original URL fails, basically just a methodology confirmation that the logout "worked".

Reply:
I guess ideally we'd have another check that if we move time 75s into the future, accessing that URL fails.  Not sure how to do that though without mocking the redis logic.  I suppose we could try changing the expiry time to 0 with `mock.patch`?  Not sure if that's worth doing, but it would validate that redis is enforcing the expiry times.

Reply:
I guess we can also verify the `url_only_url` is different from the original, like we do in the test below.